### PR TITLE
added support to convert transformer model to onnx

### PR DESCRIPTION
**Pull request by anandhu-eng**
Please note that the path to the folder that contains pytorch_model.bin should be provided along with variation.

### CODE REVIEW

File: cm-mlops/script/convert-ml-model-huggingface-to-onnx/_cm.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "deps": [
+    {
+      "names": [
+        "python3",
+        "python"
+      ],
+      "tags": "get,python3"
+    },
+    {
+      "tags": "get,generic-python-lib,_transformers,_onnxruntime"
```
Comment (arjunsuresh) [inline cm-mlops/script/convert-ml-model-huggingface-to-onnx/_cm.json:16]: We should use separate deps for each of the python packages -- the current one will only install one of the given 2.

File: cm-mlops/script/convert-ml-model-huggingface-to-onnx/run.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+python -m transformers.onnx --model=${CM_MODEL_HUGG_PATH} ${PWD}
```
Comment (arjunsuresh) [inline cm-mlops/script/convert-ml-model-huggingface-to-onnx/run.sh:2]: Can we have an input_mapping dictionary in _cm.json mapping "input":"CM_MODEL_HUGG_PATH" and in customize.py if env.get("CM_MODEL_HUGG_PATH","") == "", we can throw an error. Error can be thrown like "return {'return': 1, 'error': 'CM_MODEL_HUGG_PATH is not set'}"

Comment (anandhu-eng) [reply cm-mlops/script/convert-ml-model-huggingface-to-onnx/run.sh:2]: modified 👍🏼

File: cm-mlops/script/convert-ml-model-huggingface-to-onnx/_cm.json (hunk 1 above)
Comment (anandhu-eng) [reply cm-mlops/script/convert-ml-model-huggingface-to-onnx/_cm.json]: Please check the recent commit, i have modified the same.

File: cm-mlops/script/convert-ml-model-huggingface-to-onnx/run.sh (hunk 2 above)
Comment (arjunsuresh) [reply cm-mlops/script/convert-ml-model-huggingface-to-onnx/run.sh:2]: Thank you @anandhu-eng. Are we storing the final onnx filepath in an exported ENV variable?

Comment (anandhu-eng) [reply cm-mlops/script/convert-ml-model-huggingface-to-onnx/run.sh]: On latest commit, its being written into temp-env file as well as stored as ENV variable

File: cm-mlops/script/convert-ml-model-huggingface-to-onnx/customize.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    path = os.getcwd()
+
+    return {'return':0}
+
+def postprocess(i):
+    os_info = i['os_info']
+
+    env = i['env']
+    env['HUGGINGFACE_ONNX_FILE_PATH'] = os.path.join(os.getcwd(),"model.onnx")
+    with open('tmp-run-env.out', 'w') as f:
```
Comment (arjunsuresh) [inline cm-mlops/script/convert-ml-model-huggingface-to-onnx/customize.py:26]: Thank you @anandhu-eng for the change. Can you please remove this line? Unlike native files, inside customize.py we have direct access to env dictionary and so we don't need to use tmp-run-env.out.

Comment (anandhu-eng) [reply cm-mlops/script/convert-ml-model-huggingface-to-onnx/customize.py:26]: oh okay sure

Comment (anandhu-eng) [reply cm-mlops/script/convert-ml-model-huggingface-to-onnx/customize.py]: modified 👍🏼

**Review by arjunsuresh — APPROVED** [review]
Thank you @anandhu-eng for the fixes.

**Pull request merged**